### PR TITLE
Add authentication support 

### PR DESCRIPTION
**Pull request by setrofim**
Add support for authentication with the Veraison service. This relies on
updated apiclient that supports authentication and uses the config file
specify the authentication method and appropriate credentials.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,24 @@ The following configuration options are supported:
...
+  can be either `passthrough`/`none`, `basic`, or `oauth2`. If not specified
+  this defaults to `passthrough`. This can also be specified on the command
+  line using `-a`/`--auth` flag.
+- **username**: username for authenticating with the remote service. This is
+  only used if `auth` is set to `basic` or `oauth2`. This can also be specified
+  on the command line with `-U`/`--username`.
+- **password**: password for authenticating with the remote service. This is
+  only used if `auth` is set to `basic` or `oauth2`. This can also be specified
+  on the command line with `-P`/`--password`.
+- **client_id**: OAuth2 client ID. This used only if `auth` is set to `oauth2`.
```
Comment (thomas-fossati) [inline README.md]: s/This used/This is used/

File: README.md [hunk 2]
```diff
@@ -88,6 +88,24 @@ The following configuration options are supported:
...
+  This can also be specified on the command line using `-C`/`--client-id`.
+- **client_secret**: OAuth2 client secret. This used only if `auth` is set to
```
Comment (thomas-fossati) [inline README.md]: ditto

**Review by thomas-fossati — APPROVED** [review]
LGTM

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Pull request merged**